From 1227085c1117ee679d4b552679e9bd042942b20b Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Maxime=20M=C3=A9r=C3=A9?= <maxime.mere@foss.st.com>
Date: Fri, 30 Aug 2024 15:40:33 +0200
Subject: [PATCH 681/954] fix(stm32mp2): enable trusted boot for STM32MP21
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

STM32MP21 can use sha384 or sha256 as hash algorithm.

Set sha256 as hash algorithm by default. Sha384 support must be
developed.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 94b51a0a39831cbaa93f3d8383bdbff9a4a4a672 fix(stm32mp2): enable trusted boot for STM32MP21]
Signed-off-by: Maxime Méré <maxime.mere@foss.st.com>
Change-Id: I131ce0c6031a800b0b73d5fd9a17b71681e21b8b
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/401223
Reviewed-by: Yann GAUTIER <yann.gautier@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
---
 plat/st/stm32mp2/include/boot_api.h | 17 ++++++++---------
 1 file changed, 8 insertions(+), 9 deletions(-)

diff --git a/plat/st/stm32mp2/include/boot_api.h b/plat/st/stm32mp2/include/boot_api.h
index 53e0c72c9a..751be0875a 100644
--- a/plat/st/stm32mp2/include/boot_api.h
+++ b/plat/st/stm32mp2/include/boot_api.h
@@ -108,17 +108,10 @@
 #define BOOT_API_ECDSA_SIGNATURE_384_LEN_IN_BYTES		96
 #define BOOT_API_SHA384_DIGEST_SIZE_IN_BYTES			48
 
-#if STM32MP21
-#define BOOT_API_ECDSA_PUB_KEY_LEN_IN_BYTES			\
-	BOOT_API_ECDSA_PUB_KEY_384_LEN_IN_BYTES
-#define BOOT_API_ECDSA_SIGNATURE_LEN_IN_BYTES			\
-	BOOT_API_ECDSA_SIGNATURE_384_LEN_IN_BYTES
-#else /* !STM32MP21 */
 #define BOOT_API_ECDSA_PUB_KEY_LEN_IN_BYTES			\
 	BOOT_API_ECDSA_PUB_KEY_256_LEN_IN_BYTES
 #define BOOT_API_ECDSA_SIGNATURE_LEN_IN_BYTES			\
 	BOOT_API_ECDSA_SIGNATURE_256_LEN_IN_BYTES
-#endif /* STM32MP21 */
 
 /* Possible values of the field 'boot_api_image_header_t.ecc_algo_type' */
 #define BOOT_API_ECDSA_ALGO_TYPE_P256NIST			1
@@ -335,7 +328,11 @@ typedef struct {
 typedef struct {
 	/* BOOT_API_IMAGE_HEADER_MAGIC_NB */
 	uint32_t magic;
-	uint8_t image_signature[BOOT_API_ECDSA_SIGNATURE_LEN_IN_BYTES];
+#if STM32MP21
+	uint8_t image_signature[BOOT_API_ECDSA_SIGNATURE_384_LEN_IN_BYTES];
+#else /* STM32MP21 */
+	uint8_t image_signature[BOOT_API_ECDSA_SIGNATURE_256_LEN_IN_BYTES];
+#endif /* STM32MP21 */
 	/*
 	 * Checksum of payload
 	 * 32-bit sum all payload bytes considered as 8 bit unsigned
@@ -416,11 +413,13 @@ typedef struct {
 	 */
 	uint32_t ecc_algo_type;
 	/* ECDSA public key to be used to check signature. */
-	uint8_t ecc_pubk[BOOT_API_ECDSA_PUB_KEY_LEN_IN_BYTES];
+
 	/* table of Hash of Algo+ECDSA public key */
 #if STM32MP21
+	uint8_t ecc_pubk[BOOT_API_ECDSA_PUB_KEY_384_LEN_IN_BYTES];
 	boot_api_sha384_t pk_hashes[];
 #else /* STM32MP21 */
+	uint8_t ecc_pubk[BOOT_API_ECDSA_PUB_KEY_256_LEN_IN_BYTES];
 	boot_api_sha256_t pk_hashes[];
 #endif /* STM32MP21 */
 } __packed boot_ext_header_params_authentication_t;
-- 
2.53.0

